File: AssetManager/UserInterface/Forms/AssetManager/ViewTrackingForm.vb
Option Explicit On

Public Class ViewTrackingForm
    Sub New(parentForm As ExtendedForm, entryGUID As String, device As DeviceObject)
        InitializeComponent()
        Me.ParentForm = parentForm
        FormUID = entryGUID
        ViewTrackingEntry(entryGUID, device)
        Show()
    End Sub

    Private Sub ViewTrackingEntry(entryUID As String, device As DeviceObject)
        Try
            SetWaitCursor(True, Me)
            Dim strQry = "Select * FROM " & TrackablesCols.TableName & " WHERE  " & TrackablesCols.UID & " = '" & entryUID & "'"
            Using results As DataTable = DBFactory.GetDatabase.DataTableFromQueryString(strQry)
                For Each r As DataRow In results.Rows
                    txtTimeStamp.Text = NoNull(r.Item(TrackablesCols.DateStamp))
                    txtCheckType.Text = NoNull(r.Item(TrackablesCols.CheckType))
                    If txtCheckType.Text = "IN" Then
                        txtCheckType.BackColor = colCheckIn
                    ElseIf txtCheckType.Text = "OUT" Then
                        txtCheckType.BackColor = colCheckOut
                    End If
                    txtDescription.Text = device.Description
                    txtGUID.Text = NoNull(r.Item(TrackablesCols.DeviceUID))
                    txtCheckOutUser.Text = NoNull(r.Item(TrackablesCols.CheckoutUser))
                    txtCheckInUser.Text = NoNull(r.Item(TrackablesCols.CheckinUser))
                    txtLocation.Text = NoNull(r.Item(TrackablesCols.UseLocation))
                    txtAssetTag.Text = device.AssetTag
                    txtCheckOutTime.Text = NoNull(r.Item(TrackablesCols.CheckoutTime))
                    txtDueBack.Text = NoNull(r.Item(TrackablesCols.DueBackDate))
                    txtSerial.Text = device.Serial
                    txtCheckInTime.Text = NoNull(r.Item(TrackablesCols.CheckinTime))
                    txtNotes.Text = NoNull(r.Item(TrackablesCols.Notes))
                    txtEntryGUID.Text = NoNull(r.Item(TrackablesCols.UID))
                    Me.Text = Me.Text + " - " & NoNull(r.Item(TrackablesCols.DateStamp))
                Next
            End Using
        Catch ex As Exception
            ErrHandle(ex, System.Reflection.MethodInfo.GetCurrentMethod())
        Finally
            SetWaitCursor(False, Me)
        End Try
    End Sub

    Private Sub cmdClose_Click(sender As Object, e As EventArgs) Handles cmdClose.Click
        Me.Dispose()
    End Sub

End Class

File: AssetManager/UserInterface/Forms/AssetManagement/ViewTrackingForm.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Threading.Tasks;
using AssetManager.UserInterface.CustomControls;

namespace AssetManager.UserInterface.Forms.AssetManagement
{

	public partial class ViewTrackingForm : ExtendedForm
	{

		public ViewTrackingForm(ExtendedForm parentForm, string entryGUID, DeviceObject device)
		{
			InitializeComponent();
			this.ParentForm = parentForm;
			FormUID = entryGUID;
			ViewTrackingEntry(entryGUID, device);
			Show();
		}

		private void ViewTrackingEntry(string entryUID, DeviceObject device)
		{
			try {
				OtherFunctions.SetWaitCursor(true, this);
				var strQry = "Select * FROM " + TrackablesCols.TableName + " WHERE  " + TrackablesCols.UID + " = '" + entryUID + "'";
				using (DataTable results = DBFactory.GetDatabase().DataTableFromQueryString(strQry)) {
					foreach (DataRow r in results.Rows) {
						txtTimeStamp.Text = DataConsistency.NoNull(r[TrackablesCols.DateStamp]);
						txtCheckType.Text = DataConsistency.NoNull(r[TrackablesCols.CheckType]);
						if (txtCheckType.Text == "IN") {
							txtCheckType.BackColor = Colors.CheckIn;
						} else if (txtCheckType.Text == "OUT") {
							txtCheckType.BackColor = Colors.CheckOut;
						}
						txtDescription.Text = device.Description;
						txtGUID.Text = DataConsistency.NoNull(r[TrackablesCols.DeviceUID]);
						txtCheckOutUser.Text = DataConsistency.NoNull(r[TrackablesCols.CheckoutUser]);
						txtCheckInUser.Text = DataConsistency.NoNull(r[TrackablesCols.CheckinUser]);
						txtLocation.Text = DataConsistency.NoNull(r[TrackablesCols.UseLocation]);
						txtAssetTag.Text = device.AssetTag;
						txtCheckOutTime.Text = DataConsistency.NoNull(r[TrackablesCols.CheckoutTime]);
						txtDueBack.Text = DataConsistency.NoNull(r[TrackablesCols.DueBackDate]);
						txtSerial.Text = device.Serial;
						txtCheckInTime.Text = DataConsistency.NoNull(r[TrackablesCols.CheckinTime]);
						txtNotes.Text = DataConsistency.NoNull(r[TrackablesCols.Notes]);
						txtEntryGUID.Text = DataConsistency.NoNull(r[TrackablesCols.UID]);
						this.Text = this.Text + " - " + DataConsistency.NoNull(r[TrackablesCols.DateStamp]);
					}
				}
			} catch (Exception ex) {
				ErrorHandling.ErrHandle(ex, System.Reflection.MethodInfo.GetCurrentMethod());
			} finally {
				OtherFunctions.SetWaitCursor(false, this);
			}
		}

		private void cmdClose_Click(object sender, EventArgs e)
		{
			this.Dispose();
		}

	}
}
